### PR TITLE
feat(issue-38): use customized elements

### PR DESCRIPTION
**Pull request by jskurt**
## Issue Link

[issue-38](https://github.com/kafelix496/coding-for-fun/issues/38)

## Notes

-

### CODE REVIEW

**Review by kafelix496 — COMMENTED** [review]
1. please link `label` and `checkbox` component using `id` and `htmlFor`.
2. let's make a little bit of the space. each checkbox connected and it doesn't look good.
3. remove the resize feature at the textbox component.

File: src/modules/github/components/pull-list-by-repo/pull-list-by-repo.tsx [hunk 1]
```diff
@@ -28,38 +32,30 @@ export const PullListByRepo: FC<PullListByRepoProps> = ({
...
-          >
-            <polygon points="50,0 100,100 0,100" fill="black" />
-          </svg>
-        )}
+        <Label className="ml-2">{repo}</Label>
+        {hasChild &&
+          (isRepoOpen ? (
+            <ChevronDownIcon className={'ml-2'} onClick={handleRepoClick} />
+          ) : (
+            <ChevronRightIcon className={'ml-2'} onClick={handleRepoClick} />
```
Comment (kafelix496) [inline src/modules/github/components/pull-list-by-repo/pull-list-by-repo.tsx:45]: there is no reason to add a curly bracket here. you can just use like `className="ml-2"`

File: src/modules/github/components/pull-review-form/pull-review-form.tsx [hunk 2]
```diff
@@ -136,40 +138,35 @@ export const PullReviewForm: FC<PullReviewFormProps> = ({
...
-      <input
-        type="text"
+      <Textarea
         placeholder="Leave a comment"
-        className="w-96 h-40"
         value={commentInput}
         onChange={handleCommentChange}
         disabled={!hasChecked}
       />
       <div>
```
Comment (kafelix496) [inline src/modules/github/components/pull-review-form/pull-review-form.tsx:147]: add some space between textarea and buttons

File: src/modules/github/components/pull-review-form/pull-review-form.tsx [hunk 3]
```diff
@@ -136,40 +138,35 @@ export const PullReviewForm: FC<PullReviewFormProps> = ({
...
+        </Button>
+        <Button
           type="button"
           disabled={!hasChecked || !hasComment}
           onClick={handleRequestChangeClick}
         >
           {EPullRequestType.REQUEST_CHANGES}
-        </button>
+        </Button>
       </div>
```
Comment (kafelix496) [inline src/modules/github/components/pull-review-form/pull-review-form.tsx:170]: add some space between buttons

**Comment by jskurt** [comment]
> 1. please link `label` and `checkbox` component using `id` and `htmlFor`.
> 2. let's make a little bit of the space. each checkbox connected and it doesn't look good.
> 3. remove the resize feature at the textbox component.

1. finishied.
2. I cna't .................. help me..
3. where is it..?

File: src/modules/github/components/pull-list-by-repo/pull-list-by-repo.tsx [hunk 4]
```diff
@@ -27,39 +31,37 @@ export const PullListByRepo: FC<PullListByRepoProps> = ({
 
   return (
     <div>
-      <div className="flex">
-        <input
-          type="checkbox"
+      <div className={'flex'}>
```
Comment (kafelix496) [inline src/modules/github/components/pull-list-by-repo/pull-list-by-repo.tsx:34]: no need to use a curly bracket. just use a string.

File: src/modules/github/components/pull-list-by-repo/pull-list-by-repo.tsx [hunk 5]
```diff
@@ -27,39 +31,37 @@ export const PullListByRepo: FC<PullListByRepoProps> = ({
...
+        <Checkbox
+          id={`${repo}`}
```
Comment (kafelix496) [inline src/modules/github/components/pull-list-by-repo/pull-list-by-repo.tsx:36]: > no need to use a curly bracket. just use a string.

same here

File: src/modules/github/components/pull-list-by-repo/pull-list-by-repo.tsx [hunk 6]
```diff
@@ -27,39 +31,37 @@ export const PullListByRepo: FC<PullListByRepoProps> = ({
...
-            width="10"
-            height="10"
-            viewBox="0 0 100 100"
-            className={'ml-2' + (isRepoOpen ? ' rotate-180' : ' rotate-90')}
-            onClick={handleRepoClick}
-          >
-            <polygon points="50,0 100,100 0,100" fill="black" />
-          </svg>
-        )}
+        <Label htmlFor={`repoList.${repo}`} className="ml-2">
```
Comment (kafelix496) [inline src/modules/github/components/pull-list-by-repo/pull-list-by-repo.tsx:41]: Wrong. this `htmlFor` value must be the same as `Checkbox` component id above (line 36)

**Review by kafelix496 — CHANGES_REQUESTED** [review]

**Comment by kafelix496** [comment]
> > 1. please link `label` and `checkbox` component using `id` and `htmlFor`.
> > 2. let's make a little bit of the space. each checkbox connected and it doesn't look good.
> > 3. remove the resize feature at the textbox component.
> 
> 1. finishied.
> 2. I cna't .................. help me..
> 3. where is it..?

1. good
2. create another issue for that. https://github.com/coding-for-fun-org/frontend/issues/59. do it later
3. google it.
Screenshot 2024-01-10 at 6 46 03 AM

File: src/elements/textarea/textarea.tsx [hunk 7]
```diff
@@ -13,6 +13,7 @@ const Textarea = forwardRef<HTMLTextAreaElement, TTextareaProps>(
         )}
         ref={ref}
         {...props}
+        style={{ resize: 'none' }}
```
Comment (kafelix496) [inline src/elements/textarea/textarea.tsx:16]: check the tailwind css docs

https://tailwindcss.com/docs/resize

**Review by kafelix496 — CHANGES_REQUESTED** [review]

File: src/elements/textarea/textarea.tsx [hunk 8]
```diff
@@ -8,7 +8,7 @@ const Textarea = forwardRef<HTMLTextAreaElement, TTextareaProps>(
     return (
       <textarea
         className={clsx(
-          'flex min-h-[80px] w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50',
+          'resize-none flex min-h-[80px] w-full rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50',
```
Comment (kafelix496) [inline src/elements/textarea/textarea.tsx:11]: no, you should not touch the core element. use that class at the consuming level

**Review by kafelix496 — CHANGES_REQUESTED** [review]

**Review by kafelix496 — APPROVED** [review]
well done

**Pull request merged**